Add Dockerfile

tarcieri opened the pull request:
I'm on a Mac, and figured this would be a better way to go than attempting to build on OS X natively.

Builds a Docker container with riscv-rust-toolchain. This PR is using Debian Stretch, but if there's any distro preference I can easily change it.

This succeeded for me and I've pushed the resulting image as `tarcieri/riscv-rust-toolchain` (although I'd recommend anyone intending to use this build it for themselves).

tarcieri committed:
Add Dockerfile

Builds a Docker container with riscv-rust-toolchain

tarcieri commented on Dockerfile:
```diff
@@ -0,0 +1,54 @@
+WORKDIR /riscv-rust-toolchain
+RUN git submodule update --init --recursive
+
+ENV PATH=/riscv-rust-toolchain/toolchain/bin/:$PATH:/riscv-rust-toolchain/build/x86_64-unknown-linux-gnu/stage0/bin
```
This is the only part of it that is a little bit weird. I had to add:

```
/riscv-rust-toolchain/build/x86_64-unknown-linux-gnu/stage0/bin
```

to `$PATH` in order for `make xargo` to work. Otherwise it didn't seem to build any other version of cargo?

dvc94ch replied:
That's due to https://github.com/riscv-rust/riscv-rust-toolchain/commit/4c0c6828d38dc91e37421ee230379fa08190b059
llvm is now built as part of rust if config.toml is picked up.
Symlinking it to the root directory should work. That's also the reason why the LD_LIBRARY_PATH issue dissapeared...

dvc94ch replied:
Did you have any problems with openocd from within docker?

tarcieri replied:
Not yet. I haven't actually tried doing anything other than compiling the toolchain, though

tarcieri commented on Dockerfile:
```diff
@@ -0,0 +1,54 @@
+    zsh
+
+RUN ln -s /usr/bin/python2.7 /usr/bin/python2
+RUN git clone https://github.com/riscv-rust/riscv-rust-toolchain.git
```
Note this clones it into `/`. Perhaps it should clone it into `/root` (or perhaps even a non-root user!)

tarcieri commented:
I changed the Dockerfile to build the toolchain in `/root` (actually `$HOME`) instead of `/`, created a `$HOME/bin`, added that to `$PATH`, and symlinked `cargo` there, which seems to be working and should be a bit less hacky.

I rebuilt and pushed my build of the image with these changes. I'll let you know if I'm able to compile a project with this toolchain soon.

tarcieri commented:
I have not, but should have time to play around with it soon

jakob223 commented:
@tarcieri any updates on this?

tarcieri commented:
@jakob223 tried playing with this today and I can't get USB to work 😢 

Will close for now and consider investing in a bare metal Linux box for home
